examples/ipv6.rs

```diff
@@ -0,0 +1,126 @@
+    .get_matches();
+
+    let if_name = matches.get_one::<String>("interface").unwrap().clone();
+    let chunk_size = (*matches.get_one::<String>("chunk_size").unwrap())
+        .parse::<u32>()
+        .expect("Check chunk size.");
+    let chunk_count = (*matches.get_one::<String>("chunk_count").unwrap())
+        .parse::<u32>()
+        .expect("Check chunk count.");
+    let rx_ring_size = (*matches.get_one::<String>("rx_ring_size").unwrap())
+        .parse::<u32>()
+        .expect("Check rx ring size.");
+    let tx_ring_size = (*matches.get_one::<String>("tx_ring_size").unwrap())
+        .parse::<u32>()
+        .expect("Check tx ring size.");
+    let filling_ring_size = (*matches.get_one::<String>("filling_ring_size").unwrap())
+        .parse::<u32>()
+        .expect("Check filling ring size.");
+    let completion_ring_size = (*matches.get_one::<String>("completion_ring_size").unwrap())
+        .parse::<u32>()
+        .expect("Check completion ring size.");
+
+    /* signal define to end the application */
+    let term: Arc<AtomicBool> = Arc::new(AtomicBool::new(false));
+    let result_sigint: Result<SigId, Error> =
+        signal_hook::flag::register(signal_hook::consts::SIGINT, Arc::clone(&term));
+    let result_sigterm: Result<SigId, Error> =
+        signal_hook::flag::register(signal_hook::consts::SIGTERM, Arc::clone(&term));
+    let result_sigabrt: Result<SigId, Error> =
+        signal_hook::flag::register(signal_hook::consts::SIGABRT, Arc::clone(&term));
+
+    if result_sigint.or(result_sigterm).or(result_sigabrt).is_err() {
+        panic!("signal is forbidden");
+    }
+
+    /* execute pv_open() */
+    let pv_open_option: Option<pv::Nic> = pv::pv_open(
+        &if_name,
+        chunk_size,
+        chunk_count,
+        rx_ring_size,
+        tx_ring_size,
+        filling_ring_size,
+        completion_ring_size,
+    );
+    let mut nic: pv::Nic;
+    if let Some(a) = pv_open_option {
+        nic = a;
+    } else {
+        panic!("nic allocation failed!");
+    }
+
+    /* initialize rx_batch_size and packet metadata */
+    let rx_batch_size: u32 = 64;
+    let mut packets: Vec<pv::Packet> = Vec::with_capacity(rx_batch_size as usize);
+
+    while !term.load(Ordering::Relaxed) {
+        let received: u32 = pv::pv_receive(&mut nic, &mut packets, rx_batch_size);
+
+        if received > 0 {
+            for i in (0..received as usize).rev() {
+                unsafe {
+                    let payload_ptr = packets[i]
+                        .buffer
+                        .add(packets[i].start as usize)
+                        .cast_const();
+                    if std::ptr::read(payload_ptr.offset(12)) == 0x86 // Ethertype == 0x86DD (IPv6)
+                        && std::ptr::read(payload_ptr.offset(13)) == 0xDD
+                        && std::ptr::read(payload_ptr.offset(14)) >> 4 == 6
+                    // IP version == 6
```

Comment:
If commented line exceeded max line length, Extract it as a named variable

```suggestion
                    let ether_proto = …;
                    let ip_version = std::ptr::read(payload_ptr.offset(14)) >> 4;
                    if ether_proto == 0x86dd // IPv6
                        && ip_version == 6 // IP version
```

Or extract it as a function like `is_ipv6`